Add AmpereOne error decoding

danielf-ampere opened the pull request:
Decode AmpereOne vendor specific RAS errors.
* Add support for AmpereOne family of processors. (i.e. - ac03, ac04, and ac04_1)
* Does not contain support SQLite
* To compile RASDaemon with AmpereOne support, include the following flags: 
   --enable-arm --enable-non-standard --enable-amp-ns-decode

danielf-ampere commented:
Hi Mauro, I'm pretty new to contributing to open source projects.
Feel free to provide any and all feedback, and you can be tough on me.

Also, this change to RASDaemon benefits from this kernel changeset:
https://lore.kernel.org/all/cover.1725429659.git.mchehab+huawei@kernel.org/
Is there anything I can do to help move kernel changeset forward ?

Regards,
~Daniel Ferguson

mchehab commented:
Hi Daniel,

Please add sqlite support to it. Btw, on my long term todo plan, I'd like to change the logic to be more generic with regards to SQL, allowing it to use other databases on remote machines.

danielf-ampere commented:
Hi Mauro,

Ok, I can add SQLite support.

We have an internal patch set that adds support for MySQL, which is why I didn't add SQLite support.
So, it is a good idea to make a more generic SQL interface. I'll keep that in mind.

danielf-ampere committed:
rasdaemon: Remove static modifier from find_ns_ev_decoder.

The ras_arm_event_handler function needs to use find_ns_ev_decoder to
find the correct handler vendor specific sections inside of the standard
ARM processor error.

Signed-off-by: Daniel Ferguson <danielf@os.amperecomputing.com>

danielf-ampere committed:
rasdaemon: ampereone: Add SQLite support

This adds SQLite support for AmpereOne error decoding.

Signed-off-by: Daniel Ferguson <danielf@os.amperecomputing.com>

danielf-ampere committed:
rasdaemon: add linker flag for lib UUID

Add linker flag, and include uuid/uuid.h

Signed-off-by: Daniel Ferguson <danielf@os.amperecomputing.com>

danielf-ampere committed:
rasdaemon: Add AmpereOne vendor specific RAS error decoding.

* Add support for AmpereOne family of processors.
  (i.e. - ac03, ac04, and ac04_1)
* Does not contain support SQLite (enabled in a subsequent
  patch).
* To compile RASDaemon with AmpereOne support, include the
  following flags:
  --enable-arm --enable-non-standard --enable-amp-ns-decode

Signed-off-by: Daniel Ferguson <danielf@os.amperecomputing.com>

mchehab commented:
PR merged. I modified it to apply on the top of current upstream.

Please check.
